src/nodes/NGSI/transformer/transformer.js

```diff
@@ -0,0 +1,189 @@
+/**
+ *
+ *   NGSI Transformer Node
+ *
+ *   It transforms a NGSI(v2,LD) entity to the other NGSI(v2,LD) format.
+ *   E.g. v2 --> LD || LD --> v2
+ *
+ *   Copyright (c) 2019 Alexandra Instituttet A/S
+ *
+ *   @author Bjarke Hou Kammersgaard
+ *
+ */
+
+const URI = require("uri-js");
+
+const LD_CONTEXT = 'https://schema.lab.fiware.org/ld/context';
+const ETSI_CORE_CONTEXT = 'https://uri.etsi.org/ngsi-ld/v1/ngsi-ld-core-context.jsonld';
+
+function ngsildUri(typePart, idPart) {
+  return 'urn:ngsi-ld:'+typePart+':'+idPart;
+}
+
+function ldId(entityId, entityType) {
+  let out = entityId;
+
+  try {
+    const d = URI.parse(entityId);
+    const scheme = d.scheme;
+    if(scheme !== 'urn' && scheme !== 'http' && scheme !== 'https') {
+      throw new Error("Unknown Scheme Error");
+    }
+  } catch (e) {
+    out = ngsildUri(entityType, entityId);
+  }
+
+  return out;
+}
+
+function ldObject(attributeName, entityId) {
+  let out = entityId;
+
+  try {
+    const d = URI.parse(entityId);
+    const scheme = d.scheme;
+    if(scheme !== 'urn' && scheme !== 'http' && scheme !== 'https') {
+      throw new Error("Unknown Scheme Error");
+    }
+  } catch (e) {
+    let entityType = '';
+    if(attributeName.startsWith('ref')) {
+      entityType = attributeName.slice(3);
+    }
+
+    out = ngsildUri(entityType, entityId);
+  }
+
+  return out;
+}
+
+function ldToV2(entity) {
+  // WIP
+  return null;
+}
+
+function v2ToLD(entity, node) {
```

Comment:
it would be good to isolate this transformation and its dependencies under an independent .js module and require it in the Node code

Reply:
Sounds reasonable, I'll look into that.